VRCFaceTracking/OSC/OSCMessage.cs

```diff
@@ -1,164 +1,115 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
+using System.Runtime.InteropServices;
 using System.Text;
 
 namespace VRCFaceTracking.OSC
 {
-    public class OscMessage
-    {
-        public readonly byte[] Data;
-        public readonly string Address;
-        public readonly object Value;
+    public enum OscValueType : byte {
+        Int = 0,
+        Float = 1,
+        Bool = 2,
+        String = 3,
+    }
 
-        private OscMessage(string name, char typeIdentifier)
-        {
-            Address = name;
-            
-            var nameBytes = Encoding.ASCII.GetBytes(name);
-            nameBytes = nameBytes.EnsureCompliance();
+    [StructLayout(LayoutKind.Sequential)]
+    public struct OscValue {
+        [MarshalAs(UnmanagedType.I4)]
+        public int IntValue;
+        [MarshalAs(UnmanagedType.R4)]
+        public float FloatValue;
+        [MarshalAs(UnmanagedType.I1)]
+        public bool BoolValue;
+        [MarshalAs(UnmanagedType.LPStr)]
+        public string StringValue; // Use IntPtr for pointer types
+    }
 
-            var valueIdentBytes = Encoding.ASCII.GetBytes("," + typeIdentifier);
-            valueIdentBytes = valueIdentBytes.EnsureCompliance();
+    [StructLayout(LayoutKind.Sequential)]
+    public struct OscMessageMeta {
+        [MarshalAs(UnmanagedType.LPStr)]
+        public string Address;
+        
+        public OscValueType Type;
+        
+        public OscValue Value;
+    }
+    
+    public static class RustLib
```

Comment:
Might make sense to specify the class a bit more (RustOSCLib perhaps?), though it is under the OSC namespace.